### PR TITLE
Basic support for web wallpapers (#153)

### PR DESCRIPTION
**Pull request by Pasalc**
# Fix for issue #153. 
### Short description
Added support for simple web wallpapers(like [3026516467](https://steamcommunity.com/sharedfiles/filedetails/?id=3026516467) consisting of single html file) . If wallpaper have dependencies(like [3108493067](https://steamcommunity.com/sharedfiles/filedetails/?id=3108493067)), then program prints to user that dependencies are not supported and closes.

### CODE REVIEW

File: CMakeModules/cef_macros.cmake [hunk 1]
```diff
@@ -0,0 +1,387 @@
+# Copyright (c) 2016 The Chromium Embedded Framework Authors. All rights
```
Comment (Pasalc) [inline CMakeModules/cef_macros.cmake:1]: CopyPasted this file

File: CMakeModules/cef_variables.cmake [hunk 2]
```diff
@@ -0,0 +1,582 @@
...
+    # -Wno-error=comment              # Don't warn about code in comments
+    # -Wno-comment                    # Don't warn about code in comments
+    # -Wno-deprecated-declarations    # Don't warn about using deprecated methods
+    )
+  list(APPEND CEF_C_COMPILER_FLAGS
+    -std=c99                        # Use the C99 language standard
+    )
+  list(APPEND CEF_CXX_COMPILER_FLAGS
+    # -fno-exceptions                 # Disable exceptions
+    # -fno-rtti                       # Disable real-time type information
```
Comment (Pasalc) [inline CMakeModules/cef_variables.cmake:101]: Won't compile with this flag

File: CMakeModules/cef_variables.cmake [hunk 3]
```diff
@@ -0,0 +1,582 @@
...
+  set(CEF_LIB_RELEASE "${CEF_BINARY_DIR_RELEASE}/libcef.so")
+
+  # List of CEF binary files.
+  set(CEF_BINARY_FILES
+    chrome-sandbox
+    libcef.so
+    libEGL.so
+    libGLESv2.so
+    libvk_swiftshader.so
+    # libvulkan.so.1
```
Comment (Pasalc) [inline CMakeModules/cef_variables.cmake:222]: We don't want this file

File: main.cpp [hunk 4]
```diff
@@ -38,10 +103,18 @@ int main (int argc, char* argv[])
...
+        sLog.debug("No web wallpapers, shutting down CEF");
+        CefShutdown();
+    }
     // show the wallpaper application
     app.show ();
 
-    appPointer = nullptr;
+    if(g_CEFused){
+        sLog.debug("Shutting down CEF");
+        CefShutdown();
```
Comment (Pasalc) [inline main.cpp:109]: Shutdown CEF if we had web wallpaper

File: src/WallpaperEngine/Render/CWallpaper.cpp [hunk 5]
```diff
@@ -285,6 +351,8 @@ CWallpaper* CWallpaper::fromWallpaper (Core::CWallpaper* wallpaper, CRenderConte
         return new WallpaperEngine::Render::CScene (wallpaper->as<Core::CScene> (), context, audioContext, scalingMode);
     if (wallpaper->is<Core::CVideo> ())
         return new WallpaperEngine::Render::CVideo (wallpaper->as<Core::CVideo> (), context, audioContext, scalingMode);
-
-    sLog.exception ("Unsupported wallpaper type");
+    else if (wallpaper->is <Core::CWeb> ())
+        return new WallpaperEngine::Render::CWeb (wallpaper->as <Core::CWeb> (), context, audioContext, scalingMode);
```
Comment (Pasalc) [inline src/WallpaperEngine/Render/CWallpaper.cpp:291]: Web is now supported

File: src/WallpaperEngine/Render/CWallpaper.h [hunk 6]
```diff
@@ -16,6 +16,8 @@
 
 #include "CWallpaperState.h"
 
+#include "cefsimple_opengl/CEFGLWindow.hpp"
```
Comment (Pasalc) [inline src/WallpaperEngine/Render/CWallpaper.h]: Not needed

File: src/WallpaperEngine/Render/Drivers/CX11OpenGLDriver.cpp [hunk 7]
```diff
@@ -54,7 +54,7 @@ CX11OpenGLDriver::CX11OpenGLDriver (const char* windowTitle, CApplicationContext
 
     // make context current, required for glew initialization
     glfwMakeContextCurrent (this->m_window);
-
+
```
Comment (Pasalc) [inline src/WallpaperEngine/Render/Drivers/CX11OpenGLDriver.cpp:57]: No idea how this happened

**Comment by Almamu** [comment]
PR looks good and the backgrounds I tested seemed to work fine as long as no dependencies were required. Lacks support for multiple screens but I think this is something I can add on my own or can be done in a different PR. I'm merging it! Thanks for your contribution and sorry for the delay

**Pull request merged**